app/hypnos/src/bt.c

```diff
@@ -70,6 +74,36 @@ static struct bt_le_adv_param param = BT_LE_ADV_PARAM_INIT(
 );
 
 /* ********** Functions ********** */
+static int settings_runtime_load(void)
+{
+#if defined(CONFIG_BOARD_PINETIME)
+	settings_runtime_set("bt/dis/model",
+			     "PineTime",
+			     sizeof("PineTime"));
+	settings_runtime_set("bt/dis/manuf",
+			     "PINE64",
+			     sizeof("PINE64"));
+	settings_runtime_set("bt/dis/hw",
+			     "1.0a",
+			     sizeof("1.0a"));
+#endif
+#if defined(CONFIG_BOARD_P8)
+	settings_runtime_set("bt/dis/model",
+			     "P8",
+			     sizeof("P8"));
+	settings_runtime_set("bt/dis/manuf",
+			     "Colmi",
+			     sizeof("Colmi"));
+#endif
+	settings_runtime_set("bt/dis/sw",
```

Comment:
Could we use only this version and set `CONFIG_BT_DEVICE_NAME` and `FW_VERSION` to the correct values for the Pinetime and the P8 ?

Reply:
Do you mean to make CONFIG_BT_DEVICE_NAME depend on software revision instead of the other way around?

I don't understand what you mean by "the correct values for the Pinetime and the P8 ?"

Reply:
I meant, using this part for all 3 config (Pinetime, p8 and the other board)
```
settings_runtime_set("bt/dis/sw",
			     CONFIG_BT_DEVICE_NAME,
			     sizeof(CONFIG_BT_DEVICE_NAME));
	settings_runtime_set("bt/dis/fw",
			     FW_VERSION,
			     sizeof(FW_VERSION));

```

Reply:
I still don't understand :-) Could you show me change you want to see? Right now, all variants get values from CONFIG_BT_DEVICE_NAME and FW_VERSION, i.e. "Hypnos" and the output of "git describe --tags", respectively, while PineTime and P8 devices get additional details (copied from the PineTime wiki).

Reply:
oups, I tought it was a if elseif esle case.. Nevermind then